Added device detect for the following brands: Samsung, Sony, Tecno, MYFON, Vivax, Archos, Megafon, Manta, LG, DEXP, Irbis, Orarge, Eurostart

sanchezzzhak opened the pull request.

sanchezzzhak committed:
Detect Samsung: A40s (SM-A3051)

sanchezzzhak committed:
Detect Sony: Xperia 1 (J8110)

sanchezzzhak committed:
Detect Tecno: Camon i4 (CB7j)

sanchezzzhak committed:
Detect Tecno: Camon i4 (CB7)

sanchezzzhak committed:
Detect MYFON: My Pad 7s (MYPAD7s)

sanchezzzhak committed:
Detect Vivax: Fly 3, Fly 5 Lite, Smart Point X5010

sanchezzzhak committed:
#6021 Detect Archos: Oxygen 63

sanchezzzhak commented:
wow fast tests passed :+1:

sanchezzzhak committed:
#6041 Detect MegaFon: Optima (MS3B)

sanchezzzhak committed:
#6041 Detect MegaFon: Login 4 LTE (MFLogin4)

sanchezzzhak committed:
#6041 Detect MegaFon: Mint (SP-A20i)

sanchezzzhak committed:
#6041 Detect MegaFon: SP-A5 (SP-A5)

sanchezzzhak committed:
#6041 Detect Manta: Titano 3 (MSP95014)

sanchezzzhak committed:
#6041 Detect Manta: Mezo 2 (MSP95015)

sanchezzzhak committed:
#6041 Detect LG: G5 (VS987)

sanchezzzhak committed:
#6041 Detect LG: Stylo 2 V (VS835)

sanchezzzhak committed:
#6041 Detect LG: K8 V (VS500)

sanchezzzhak committed:
#6041 Detect DEXP: BL150

sanchezzzhak committed:
#6041 BQ BQ-7000G Сharm
#6041 BQ BQ-7098G Armor Power
#6041 DEXP BS155
#6041 DEXP BS160
#6041 DEXP BS550
#6041 DEXP BS650
#6041 Venso CX-508

sanchezzzhak committed:
#6041 Irbis: TZ794, TZ831, TZ841, TZ777, TZ771, TZ150, TZ165, TZ175, TZ176, TZ177, TZ178
#6041 Philips: TLE722G
#6041 Manta: Rocky 2 MSP95020, Rocky 3 Pro MSP95021, Forto 2 MSP96017

sanchezzzhak committed:
#6041 Orange: Rise 34, Dive72, Rise51

sgiehl commented on regexes/device/mobiles.yml:
```diff
@@ -7835,7 +7863,7 @@ Orange:
       model: 'FunTab $1'
       device: 'tablet'
-    - regex: 'Orange[ _-]([^/;]+) Build'
+    - regex: 'Orange[ _-]([^/;]+)( Build|\))'
```
When using `)` as "stop" character, it should be excluded from the match pattern, as it otherwise could match to much.
Would imho be better to use this
```suggestion
    - regex: 'Orange[ _-]([^/;)]+)( Build|\))'
```

sanchezzzhak committed:
fix: tests orange fixed

sanchezzzhak committed:
feat(Eurostar): added detect new brand
detect devices Onyx 1S, Onyx 3S, ePad 4S
issue #6041

sanchezzzhak committed:
Merge branch 'master' into pr_device_23

mimmi20 commented on regexes/device/mobiles.yml:
```diff
@@ -7823,7 +7862,7 @@ Opsson:
 # orange
 Orange:
-  regex: 'SPV[ \-]?([a-z0-9]+)|Orange[ _-]([^/;]+) Build|SC/IHD92|FunTab ([^/;]+) Build'
+  regex: 'SPV[ \-]?([a-z0-9]+)|Orange[ _-]([^/;]+)( Build|\))(?<!Tahiti-LS\))|SC/IHD92|FunTab ([^/;]+) Build'
```
Schould the term `(?<!Tahiti-LS\))` not be before the searched word, not behind?

sanchezzzhak replied:
I tried to put in the beginning but it didn't work
success https://regex101.com/r/PaZaPd/1
bad https://regex101.com/r/PaZaPd/2

mimmi20 replied:
I would write the regex this way: `Orange[ _-](?!Tahiti-LS\))([^\/;]+)( Build|\))`

sanchezzzhak replied:
Thanks

sanchezzzhak committed:
fix: resolve conflict merge

sanchezzzhak committed:
feat(devices): added detect devices
Sony Xperia L3 (I3312), Xperia L3 Dual (I4312, I4332)
LG X4 (2019) (LM-X420N)
Bravis Space (N1-570 Space)
Oppo A7x (PBBM00, PBBT00), A7n (PCDM00), A3s (PBBM30), R15x (PBCM10), R17 (PBET00)
Huawey Honor 6 Plus (PE-UL00), MediaPad M5 lite (BAH2-L09, BAH2-W19)
issue #6041

sanchezzzhak committed:
fix: PR Orange common regexp fixed

sanchezzzhak committed:
feat(devices): added detect devices
Tooky A19, W1, T86, T1982, T83, T88, A9PLUS
issue #6041

sanchezzzhak committed:
feat(devices): added detect devices
Oppo Realme 3i (RMX1827), Realme X (RMX1901), Realme 5 (RMX1911), RealMe C2 (RMX1943), Realme 5 Pro (RMX1971)
issue #6041

sanchezzzhak committed:
feat(devices): added detect devices
Xiaomi Black Shark 2 (SKW-A0)
Sony Xperia XZ1 (SO-01K)
Huawei MediaPad M6 8.4 (VRD-AL09), Honor Y6 II Compact (LYO-L21)
Asus Max Pro M1 (ZB601KL)
Beeline Pro 2
Vivo Y93 (V1818A)
issue #6041
